### ISSUE TITLE
`@stylistic/declaration-block-trailing-semicolon` offers a fix a Sass nested property never receives, and the warning goes with it

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#335]
Found by the review of the branch closing [#287](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/287), and older than it: this shape reports and swallows its own warning on `main`, and on 5.3.0 before it.

### What minimal example or steps are needed to reproduce the bug?

```scss
a {
	font: 12px
	{ family: serif; }
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "customSyntax": "postcss-scss",
  "rules": {
    "@stylistic/declaration-block-trailing-semicolon": "always"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --config myconfig.json --fix
```

### What did you expect to happen?

Either the semicolon written, or the warning left standing over code the fix cannot reach. Without `--fix` the warning is reported, at `2:11`.

### What actually happened?

With `--fix` the run ends clean and not a byte is written. The problem is reported, the fix callback is called, and Stylelint takes the problem for solved — so the file still violates the rule and nothing says so. A comment behind the value changes nothing: `font: 12px /* c */` goes the same way.

### What is the cause?

A Sass nested property is a declaration carrying a block, and `postcss-scss` is the only syntax that reads one — `postcss-less` and plain CSS read those three lines as a rule whose selector is `font: 12px`. The fix of this rule writes no character of its own; it sets `parent.raws.semicolon` and leaves the printing to the stringifier. `ScssStringifier.decl` prints such a declaration as its head, its block and the closing brace, and drops the `semicolon` argument it was handed — exactly as PostCSS's own `atrule` drops it for an at-rule carrying a block. So the flag speaks of a semicolon nothing will print.

The at-rule half of that is answered already: `root.walkAtRules` turns away every at-rule for which `hasBlock` is true. The declaration walk three lines below asks no such thing, so a nested property reaches `checkLastNode` and is offered a fix that cannot land.

Whether such a declaration should be asked about at all is the wider question — Sass compiles the block with a semicolon behind it and without one, so `always` has something to ask for, and only the writing of it is out of reach.
